raft/src/test/java/org/apache/kafka/raft/MockLog.java

```diff
@@ -316,6 +316,11 @@ public void flush() {
         lastFlushedOffset = endOffset().offset;
     }
 
+    @Override
+    public boolean maybeClean() {
+        return false;
+    }
```

Comment:
Are you planning to implement this and use this instead of `deleteBeforeSnapshot` below? Most of the tests in the `raft` module depend on `MockLog` having the same semantic as `KafkaMetadataLog`.

Reply:
I'm going to address this further in a follow-up PR